crates/commitlog/src/segment.rs

```diff
@@ -411,15 +430,17 @@ mod tests {
     fn metadata() {
         let repo = repo::Memory::default();
 
-        let mut writer = repo::create_segment_writer(&repo, Options::default(), 0).unwrap();
-        writer.append([0; 32]).unwrap();
-        writer.append([0; 32]).unwrap();
-        writer.commit().unwrap();
-        writer.append([1; 32]).unwrap();
-        writer.commit().unwrap();
-        writer.append([2; 32]).unwrap();
-        writer.append([2; 32]).unwrap();
-        writer.commit().unwrap();
+        {
```

Comment:
Should move those blocks back to the left -- shouldn't depend on drop, actually.